### DISCUSSION TITLE
Capital 'R' file extension not recognized

### DISCUSSION THREAD

**Discussion by PMassicotte** [kyazdani42/nvim-web-devicons#504] [General]
In R, it is standard practice to use a capital "R" when naming files, such as `test.R`. However, this seems to cause an issue in this context, as files with this naming convention are not being recognized. Changing the extension to `.r` resolves the issue (see the attached screenshot).

![image](https://github.com/user-attachments/assets/292089a0-00d2-4b76-9d86-07af0805b2df)

I suspect this behaviour may stem from this function:

https://github.com/nvim-tree/nvim-tree.lua/blob/010ae0365aafd6275c478d932515d2e8e897b7bb/lua/nvim-tree/renderer/components/icons.lua#L57-L75)

Is there anything I can do on my end to modify this behavior?

**Comment by PMassicotte** [comment]
Using 

```lua
    extension = string.lower(extension)
```
in the function fixes the problem. It comes from the fact that `devicons` use lowercases for lookup:

https://github.com/nvim-tree/nvim-web-devicons/blob/56f17def81478e406e3a8ec4aa727558e79786f3/lua/nvim-web-devicons.lua#L231

https://github.com/nvim-tree/nvim-web-devicons/blob/56f17def81478e406e3a8ec4aa727558e79786f3/lua/nvim-web-devicons.lua#L518-L521

**Comment by gegoune** [comment]
Should be fixed by https://github.com/nvim-tree/nvim-web-devicons/pull/503.

**Comment by PMassicotte** [comment]
I am using the latest commit of https://github.com/nvim-tree/nvim-web-devicons and still have this issue. devicons lowercase the name of the file, not its extension.

**Reply by alex-courtis** [reply]
Should be fixed, extension has always been missing.

#505
